manifests/check.pp

```diff
@@ -129,89 +129,39 @@
 #   of the Hash value.
 #   Default: undef
 define sensu::check(
-  $command,
-  $ensure              = 'present',
-  $type                = undef,
-  $handlers            = undef,
-  $contacts            = undef,
-  $standalone          = true,
-  $cron                = 'absent',
-  $interval            = 60,
-  $occurrences         = undef,
-  $refresh             = undef,
-  $source              = undef,
-  $subscribers         = undef,
-  $low_flap_threshold  = undef,
-  $high_flap_threshold = undef,
-  $timeout             = undef,
-  $aggregate           = undef,
-  $aggregates          = undef,
-  $handle              = undef,
-  $publish             = undef,
-  $dependencies        = undef,
-  $custom              = undef,
-  $ttl                 = undef,
-  $subdue              = undef,
-  $proxy_requests      = undef,
+  String $command,
+  Enum['present','absent'] $ensure              = 'present',
+  Optional[String] $type                = undef,
+  Variant[Undef,String,Array] $handlers            = undef,
+  Optional[Array] $contacts            = undef,
+  Variant[Boolean,Enum['absent']] $standalone          = true,
+  String $cron                = 'absent',
+  Variant[Integer,Enum['absent']] $interval            = 60,
+  Variant[Undef,Pattern[/^(\d+)$/],Integer,Enum['absent']] $occurrences         = undef,
```

Comment:
RFC: In the module in most parts, numbers as accepted both as string and numbers. I've used this approach (Pattern[/^(\d+)$/],Integer ) in these cases, there could be something better maybe.